src/group_theory/subgroup.lean

```diff
@@ -1406,6 +1406,17 @@ le_antisymm
     (gclosure_preimage_le _ _))
   ((closure_le _).2 $ set.image_subset _ subset_closure)
 
+-- this instance can't go just after the definition of mrange because `fintype` is
+-- not imported at that stage
+@[to_additive]
+instance fintype_mrange {M N : Type*} [monoid M] [monoid N] [fintype M] [decidable_eq N]
+  (f : M →* N) : fintype (mrange f) :=
+set.fintype_range f
+
+@[to_additive]
```

Comment:
```suggestion
/-- Note: this can form a diamond with `subgroup.fintype` in the presence of `fintype N`. -/
@[to_additive /-" Note: this can form a diamond with `add_subgroup.fintype` in the presence
of `fintype N`. "-/]
```

Reply:
Neither seem to be there?

Reply:
On poking around further I realise that they're all `subtype.fintype` e.g.
```
example (F : Type) [field F] [fintype F] (E : subfield F) [decidable_pred (∈ E)] :
  fintype E := infer_instance -- subtype.fintype (λ (x : F), x ∈ E)
```